src/main/java/net/glowstone/io/anvil/RegionFile.java

```diff
@@ -200,19 +214,19 @@ public DataInputStream getChunkDataInputStream(int x, int z) throws IOException
         if (version == VERSION_GZIP) {
             byte[] data = new byte[length - 1];
             file.read(data);
-            return new DataInputStream(new GZIPInputStream(new ByteArrayInputStream(data)));
+            return new DataInputStream(new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(data))));
```

Comment:
`InflaterInputStream` and `GZIPInputStream` do have a buffer for reading from an input stream. However, they are used for different purposes: `InflaterInputStream` uses it to ensure that data is read from the backing `InputStream`, whereas the `BufferedInputStream` is meant to buffer the decompressed data. As a side effect, this reduces the number of inflations required to decompress the chunk.